### PR TITLE
Added test cases for Metrics

### PR DESCRIPTION
**Pull request by asaharn**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/azuredataexplorerexporter/metricsdata_to_adx_test.go [hunk 1]
```diff
@@ -192,6 +192,180 @@ func Test_mapToAdxMetric(t *testing.T) {
...
+					MetricName: "Int_gauge_value",
+					MetricType: "Gauge",
+					Value:      5,
+					Host:       "test-host",
+					Attributes: `{"key":"value"}`,
+				},
+			},
+		},
+		{
+			name: "Float_gauge_value",
```
Comment (ag-ramachandran) [inline exporter/azuredataexplorerexporter/metricsdata_to_adx_test.go:239]: lets lower case all the test names

File: exporter/azuredataexplorerexporter/metricsdata_to_adx_test.go [hunk 2]
```diff
@@ -192,6 +192,180 @@ func Test_mapToAdxMetric(t *testing.T) {
...
+		},
+		{
+			name: "unknown_type",
+			resourceFn: func() pcommon.Resource {
+				return newMetricsWithResources()
+			},
+			metricsDataFn: func() pmetric.Metric {
+				metric := pmetric.NewMetric()
+				metric.SetName("unknown_with_dims")
+				metric.SetDataType(pmetric.MetricDataTypeNone)
```
Comment (ag-ramachandran) [inline exporter/azuredataexplorerexporter/metricsdata_to_adx_test.go:361]: can we add a test if this is nil as well

File: exporter/azuredataexplorerexporter/metricsdata_to_adx_test.go [hunk 3]
```diff
@@ -192,6 +192,180 @@ func Test_mapToAdxMetric(t *testing.T) {
...
+				gauge.SetName("nil_gauge_value")
+				gauge.SetDataType(pmetric.MetricDataTypeGauge)
+				return gauge
+			},
+			configFn: func() *Config {
+				return createDefaultConfig().(*Config)
+			},
+		},
+		{
+			name: "Int_gauge_value",
```
Comment (ag-ramachandran) [inline exporter/azuredataexplorerexporter/metricsdata_to_adx_test.go:211]: just as below , lets have all lower case test names
